boms/cloud-oss-bom/pom.xml

```diff
@@ -126,8 +126,6 @@
       </dependency>
 
       <!-- GRPC; specifically https://github.com/grpc/grpc-java -->
-      <!-- We should import the officially-provided grpc BOM once it is created:
-           grpc/grpc-java#3954 -->
```

Comment:
Will you replace this with why we're not importing the gRPC BOM?

Reply:
I tired that and it broke the tests. MIght be worth digging further, but it seems the grpc BOM pulls in things we don't currenlty have in libraries-bom.

Reply:
Let me try to reproduce the problem.

Reply:
grpc-bom includes an artifact that is not published: `io.grpc:protoc-gen-grpc-java:jar:1.23.0`. Would you update the comment accordingly?

```
suztomo@suxtomo24:~/cloud-opensource-java/dashboard$  mvn exec:java -Dexec.arguments="-f ../boms/cloud-oss-bom/pom.xml"
[INFO] Scanning for projects...
[INFO] 
[INFO] ------------------< com.google.cloud.tools:dashboard >------------------
[INFO] Building Cloud Tools Open Source Code Hygiene Dashboard 1.0.1-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- exec-maven-plugin:1.6.0:java (default-cli) @ dashboard ---
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
[WARNING] 
org.eclipse.aether.resolution.DependencyResolutionException: Could not find artifact io.grpc:protoc-gen-grpc-java:jar:1.23.0 in central (http://repo1.maven.org/maven2/)
    at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies (DefaultRepositorySystem.java:352)
    at com.google.cloud.tools.opensource.dependencies.DependencyGraphBuilder.resolveCompileTimeDependencies (DependencyGraphBuilder.java:162)
    at com.google.cloud.tools.opensource.dependencies.DependencyGraphBuilder.getStaticLinkageCheckDependencyGraph (DependencyGraphBuilder.java:191)
    at com.google.cloud.tools.opensource.classpath.ClassPathBuilder.artifactsToDependencyPaths (ClassPathBuilder.java:86)
    at com.google.cloud.tools.opensource.dashboard.DashboardMain.generate (DashboardMain.java:182)
    at com.google.cloud.tools.opensource.dashboard.DashboardMain.generate (DashboardMain.java:171)
    at com.google.cloud.tools.opensource.dashboard.DashboardMain.main (DashboardMain.java:108)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.mojo.exec.ExecJavaMojo$1.run (ExecJavaMojo.java:282)
    at java.lang.Thread.run (Thread.java:748)
Caused by: org.eclipse.aether.resolution.ArtifactResolutionException: Could not find artifact io.grpc:protoc-gen-grpc-java:jar:1.23.0 in central (http://repo1.maven.org/maven2/)
    at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve (DefaultArtifactResolver.java:423)
    at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts (DefaultArtifactResolver.java:225)
    at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies (DefaultRepositorySystem.java:335)
    at com.google.cloud.tools.opensource.dependencies.DependencyGraphBuilder.resolveCompileTimeDependencies (DependencyGraphBuilder.java:162)
    at com.google.cloud.tools.opensource.dependencies.DependencyGraphBuilder.getStaticLinkageCheckDependencyGraph (DependencyGraphBuilder.java:191)
    at com.google.cloud.tools.opensource.classpath.ClassPathBuilder.artifactsToDependencyPaths (ClassPathBuilder.java:86)
    at com.google.cloud.tools.opensource.dashboard.DashboardMain.generate (DashboardMain.java:182)
    at com.google.cloud.tools.opensource.dashboard.DashboardMain.generate (DashboardMain.java:171)
    at com.google.cloud.tools.opensource.dashboard.DashboardMain.main (DashboardMain.java:108)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.mojo.exec.ExecJavaMojo$1.run (ExecJavaMojo.java:282)
    at java.lang.Thread.run (Thread.java:748)
Caused by: org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact io.grpc:protoc-gen-grpc-java:jar:1.23.0 in central (http://repo1.maven.org/maven2/)
    at org.eclipse.aether.connector.basic.ArtifactTransportListener.transferFailed (ArtifactTransportListener.java:48)
    at org.eclipse.aether.connector.basic.BasicRepositoryConnector$TaskRunner.run (BasicRepositoryConnector.java:369)
    at org.eclipse.aether.util.concurrency.RunnableErrorForwarder$1.run (RunnableErrorForwarder.java:75)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1149)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:624)
    at java.lang.Thread.run (Thread.java:748)

```

Reply:
I;nm ont sure that's it. I see it here: https://repo1.maven.org/maven2/io/grpc/protoc-gen-grpc-java/1.23.0/protoc-gen-grpc-java-1.23.0.pom

Reply:
The problem is the artifact packaging is pom while our dashboard is trying to download a JAR file.
Created https://github.com/GoogleCloudPlatform/cloud-opensource-java/issues/943 .

Reply:
Thanks for tracking this down. I think we're better off tracking the issue in the bug tracker rather than file comments now that we have a grip on what's happening.